server/src/main/java/org/opensearch/common/cache/tier/TierType.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.tier;
+
+/**
+ * Tier types in cache.
+ */
+public enum TierType {
+
+    ON_HEAP,
+    DISK;
```

Comment:
OffHeap?

Reply:
As it is not used for now, we can add it later when needed.

Reply:
Does it still make sense to have ON_HEAP / OFF_HEAP as primary tiers and DISK as one of the OFF_HEAP sub tiers?

Reply:
I have renamed this CacheStoreTypes as per @sohami suggestion. I think having ON_HEAP, DISK make sense.